src/operator/contrib/csr_neighborhood_sample-inl.h

```diff
@@ -213,35 +240,60 @@ static void CSRNeighborSampleComputeExCPU(const nnvm::NodeAttrs& attrs,
   }
 
   std::vector<dgl_id_t> tmp_src_list;
-  std::vector<dgl_id_t> tmp_sampled_list;
+  std::vector<dgl_id_t> tmp_edge_list;
+  std::vector<dgl_id_t> tmp_sampled_src_list;
+  std::vector<dgl_id_t> tmp_sampled_edge_list;
+
+  std::map<dgl_id_t, std::vector<dgl_id_t> > ver_mp;
+  std::map<dgl_id_t, std::vector<dgl_id_t> > edge_mp;
 
-  while (!node_queue.empty() && sub_vertices_count < max_num_vertices) {
+  while (!node_queue.empty()) {
     ver_node& cur_node = node_queue.front();
     if (cur_node.level < num_hops) {
+
       dgl_id_t dst_id = cur_node.vertex_id;
       tmp_src_list.clear();
-      tmp_sampled_list.clear();
-      GetSrcList(col_list, indptr, dst_id, tmp_src_list);
-      GetSample(tmp_src_list, num_neighbor, tmp_sampled_list);
-      for (size_t i = 0; i < tmp_sampled_list.size(); ++i) {
-        auto got = sub_ver_mp.find(tmp_sampled_list[i]);
+      tmp_edge_list.clear();
+      tmp_sampled_src_list.clear();
+      tmp_sampled_edge_list.clear();
+
+      GetSrcList(val_list, 
+                 col_list, 
+                 indptr, 
+                 dst_id, 
+                 tmp_src_list, 
+                 tmp_edge_list);
+
+      GetSample(tmp_src_list, 
+                tmp_edge_list, 
+                num_neighbor, 
+                tmp_sampled_src_list,
+                tmp_sampled_edge_list);
+
+      ver_mp[dst_id] = tmp_sampled_src_list;
+      edge_mp[dst_id] = tmp_sampled_edge_list;
+      
+      sub_vertices_count++;
+      if (sub_vertices_count == max_num_vertices) {
```

Comment:
all sampled neighbors in `src_list` are the vertices in the sampled subgraph. We need another map to trace the number of unique elements in the subgraph.